flink/v1.19/flink/src/test/java/org/apache/iceberg/flink/SqlBase.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.flink;
+
+import static org.apache.iceberg.flink.FlinkCatalogFactory.DEFAULT_CATALOG_NAME;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+import java.util.Map;
+import org.apache.flink.table.api.TableEnvironment;
+import org.apache.flink.table.api.TableResult;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.CloseableIterator;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+
+public abstract class SqlBase {
+  protected abstract TableEnvironment getTableEnv();
+
+  protected static TableResult exec(TableEnvironment env, String query, Object... args) {
+    return env.executeSql(String.format(query, args));
+  }
+
+  protected TableResult exec(String query, Object... args) {
+    return exec(getTableEnv(), query, args);
+  }
+
+  protected List<Row> sql(String query, Object... args) {
+    TableResult tableResult = exec(query, args);
+    try (CloseableIterator<Row> iter = tableResult.collect()) {
+      return Lists.newArrayList(iter);
+    } catch (Exception e) {
+      throw new RuntimeException("Failed to collect table result", e);
+    }
+  }
+
+  protected void assertSameElements(Iterable<Row> expected, Iterable<Row> actual) {
+    assertThat(actual).isNotNull().containsExactlyInAnyOrderElementsOf(expected);
+  }
+
+  protected void assertSameElements(String message, Iterable<Row> expected, Iterable<Row> actual) {
+    assertThat(actual).isNotNull().as(message).containsExactlyInAnyOrderElementsOf(expected);
+  }
+
+  /**
+   * We can not drop currently used catalog after FLINK-29677, so we have make sure that we do not
+   * use the current catalog before dropping it. This method switches to the 'default_catalog' and
+   * drops the one requested.
+   *
+   * @param catalogName The catalog to drop
+   * @param ifExists If we should use the 'IF EXISTS' when dropping the catalog
+   */
+  protected void dropCatalog(String catalogName, boolean ifExists) {
```

Comment:
Can we do a throughout cleanup instead of dropping random catalogs?

Like:
- Dropping every catalog other than the dafault
- Dropping every database other than the default
- Dropping evey table other than the default?

Reply:
right now, inherited classes clean up created tables and databases explicitly, e.g. `TableSourceTestBase`. We can implement what you suggested. but it will be a couple more SQL executions (1) list databases (2) for each database (usually one), list tables. I am leaning a little more toward the existing approach.

```
  @AfterEach
  public void clean() {
    sql("DROP TABLE IF EXISTS %s.%s", DATABASE_NAME, TABLE_NAME);
    dropDatabase(DATABASE_NAME, true);
    dropCatalog(CATALOG_NAME, true);
  }
```

Reply:
I think it is good to reduce the complexity of writing tests.
If we create a good cleanup method we can use everywhere then we just call that in an `AfterEach` method and forget about it altogether. Otherwise we need to remember which database/table was created during the tests. Also, if you different test methods creating different tables, then you need to add logic to the cleanup to differentiate those.

I feel more inclined to the generic `cleanup` method, even if it adds a few more calls to the test runs.

What do you think @rodmeneses?

Reply:
> even if it adds a few more calls to the test runs.

note that this is for each test method. with parameterized tests, some classes have dozens of tests. this can add up quickly.

Reply:
Do we have numbers?
Higher test cases means higher probability of a mistake to forget to delete a db or a table.

Reply:
if we want to redo the catalog clean up pattern, I think it is better to do it in a separate PR if we can get consensus on the new approach.